### ISSUE TITLE
Windows: The machinery to emulate app in tests is broken

### ISSUE THREAD

**Issue by BoboTiG** [pyinstaller/pyinstaller#3590]
With the issue #3520, we discovered that the machinery to emulate a compiled app is partly broken.

For instance, let's try the `test_PyQt5_SSL_support()` test function from `test_libraries.py`. Before the patch was merged, it should have failed. But it never failed because the content of `os.environ` when running the test was setted to a location where desired DLLs were found:
```python
# ...
'PATH': 'c:\\python-3.6.5\\lib\\site-packages\\PyQt5\\Qt\\bin;...',
``` 
DLLs are present into the path and the test did not fail.

Printing its value from the manually packaged app:
```python
# ...
'PATH': 'C:\\Users\\TestW764\\AppData\\Local\\Temp\\_MEI23362\\PyQt5\\Qt\\bin;...'
```
And there, no DLLs: failure.

It may have several tests that we cannot trust until a fix is found.

**Comment by bjones1** [comment]
The problem occurs at [prog_env['PATH'] = os.pathsep.join(winutils.get_system_path())](https://github.com/pyinstaller/pyinstaller/blob/develop/tests/functional/conftest.py#L326), which calls [get_system_path](https://github.com/pyinstaller/pyinstaller/blob/develop/PyInstaller/utils/win32/winutils.py#L40), which calls [compat.getenv](https://github.com/pyinstaller/pyinstaller/blob/develop/PyInstaller/utils/win32/winutils.py#L54), which is defined [here](https://github.com/pyinstaller/pyinstaller/blob/develop/PyInstaller/compat.py#L278) and just calls `os.environ.get`.

I would think removing [line 54](https://github.com/pyinstaller/pyinstaller/blob/develop/PyInstaller/utils/win32/winutils.py#L54) would fix this, but I'm low on time to play with this.

**Comment by BoboTiG** [comment]
Then I will have a look soon :)

**Comment by bjones1** [comment]
Closed by b15a42736ef6e8c24cb3e449186f4ac2c5df96fc.
